client/src/main/java/io/pravega/client/tables/impl/KeyVersion.java

```diff
@@ -9,16 +9,45 @@
  */
 package io.pravega.client.tables.impl;
 
+import java.io.Serializable;
 import java.nio.ByteBuffer;
 
 /**
  * Version of a Key in a Table.
  */
-interface KeyVersion {
+public interface KeyVersion extends Serializable {
+
     /**
      * A special KeyVersion which indicates the Key must not exist when performing Conditional Updates.
```

Comment:
This is for executing conditional updates. If you do a conditional update and pass this as the compare version, then you are conditioning on the fact that the key does not exist. In other words, it is an "insert", which would fail if the key already exists.

The Wire Protocol defines two responses for this: one in the case when the Key does not exist and one when the update failed because of a bad version.